### ISSUE TITLE
Strategy for handling policies/sets that are Cloud-type dependent

### ISSUE THREAD

**Issue by gregslack78** [Azure/enterprise-azure-policy-as-code#311]
Some built in policies/policySets do not exist in China (and presumably Gov).  Because of this a plan may fail when evaluating a policySet and comes across a reference to a built in policy that doesn't exist in the tenant being deployed to.  Additionally, when assignments are evaluated they evaluate the validity of the policy object in the assignment (does that policy/policySet exist in the tenant I am about to deploy to?) even when the assignment is not being made as part of the current deployment.

**Comment by techlake** [comment]
@gregslack78:  please implement this. I changed my mind. using filename convention for definitions and assignments is good:
xyz.ChinaCloud.json
don't allow multiples (this is not that common a scenario)
But add: xyz.-.json for any not explicitly called out

**Comment by gregslack78** [comment]
Working on it right now but doing it with the metadata approach.  Is there a reason the file name approach would be preferred?

I've currently got the deploy part working where it will skip a file if it has

    "metadata": {
      "EPaCCloudEnvironments": [
        "AzureChinaCloud"  //only deploy to AzureChina, skip all other cloud types.
      ]
     }


________________________________

**Comment by techlake** [comment]
OK. Minor change requests:

- Spelling should be camelCase for property names -> "epacCloudEnvironment"
- Should we instead of cloud use the epacEnvironment (since it already defines which cloud we are operating in)
- We could use this for Policy/Initiative definitions
- And fix the order of checking in Assignments
Thoughts?

**Comment by gregslack78** [comment]
*   Spelling should be camelCase for property names -> "epacCloudEnvironment" - Sounds good.  I'll make that change
  *   Should we instead of cloud use the epacEnvironment (since it already defines which cloud we are operating in) - if we went with epacEnvironment, we could potentially have to list A LOT of those.  Yes, it would allow for granularity, but in this case I don't think we want that.  I think listing the cloud type is the right thing to do here since the point is to say
  *   We could use this for Policy/Initiative definitions - I think I've got this working for policies, inits, and assignments
  *   And fix the order of checking in Assignments
Thoughts?

The only thing I am stuck on is the scenario where we have a policy that doesn't have this metadata property, so it is deploying to all clouds.  Then, I decide I only want that policy going to China.  I now need that policy to be deleted from the other cloud types.  I'm having a hard time figuring out how to leave this in the deleteCandidates hash table.

________________________________
